lib/components/Input/generic-wheel-picker/index.tsx

```diff
@@ -0,0 +1,216 @@
+
+        return (
+            <>
+                <div
+                    ref={containerRef}
+                    className={clsx(
+                        "quill-generic-picker__container",
+                        containerClassName,
+                    )}
+                >
+                    <div
+                        ref={ref}
+                        onClick={handleInputClick}
+                        onKeyDown={handleKeyDownEvent}
+                        onMouseDown={handleMouseDown}
+                        onMouseUp={handleMouseUp}
+                        onMouseLeave={handleMouseLeave}
+                    >
+                        <Input
+                            dropdown
+                            isDropdownOpen={isOpen}
+                            value={selectedValue ?? initialValues}
+                            autoFocus={false}
+                            autoComplete="off"
+                            readOnly={true}
+                            className={clsx(
+                                "quill-generic-picker__input",
+                                `quill-generic-picker__input--is-open-${isOpen && !isPressed}--${rest.variant}--${rest.status}`,
+                                className,
+                            )}
+                            type="text"
+                            label={label}
+                            leftIcon={leftIcon}
+                            rightIcon={rightIcon}
+                            {...rest}
+                        />
+                    </div>
+                    <div className="quill-generic-picker__content--container">
+                        {!isMobile ? (
+                            isOpen && (
+                                <div className="quill-generic-picker__content">
+                                    <WheelPickerContainer
+                                        inputValues={inputValues}
+                                        close={close}
+                                        setSelectedValue={setSelectedValue}
+                                        setInputValues={(
+                                            index: number,
+                                            newValue: string | number,
+                                        ) => updateItemAtIndex(index, newValue)}
+                                        {...rest}
+                                    >
+                                        {children}
+                                    </WheelPickerContainer>
+                                </div>
+                            )
+                        ) : (
+                            <ActionSheet.Root isOpen={isOpen} onClose={close}>
+                                <ActionSheet.Portal
+                                    shouldCloseOnDrag={true}
+                                    fullHeightOnOpen={true}
+                                    ref={actionSheetRef}
+                                >
```

Comment:
@jia-deriv looks like `isMobile` hook achieves thesame purpose of the `breakpointProvider` can you confirm?